### PR TITLE
fix(browse): Use a simpler approach for representing paths responsively

### PR DESCRIPTION
**Pull request by rafpaf**
### Description

The existing `CollectionBreadcrumbsWithTooltip` component was clunky and unperformant. This PR introduces a simpler, CSS-only approach.

### How to verify

Go to **Browse models** and look at the collection column.

### Checklist

- [ ] Tests have been added/updated to cover changes in this PR - since this is just CSS, I haven't added any tests

### CODE REVIEW

File: frontend/src/metabase/browse/components/ModelsTable.tsx [hunk 1]
```diff
@@ -81,11 +82,9 @@ export const ModelsTable = ({
   models = [],
   skeleton = false,
 }: ModelsTableProps) => {
-  // for large datasets, we need to simplify the display to avoid performance issues
   const isLargeDataset = models.length > LARGE_DATASET_THRESHOLD;
 
-  const [showLoadingManyRows, setShowLoadingManyRows] =
-    useState(isLargeDataset);
+  const [showLoadingManyRows, setShowLoadingManyRows] = useState(false);
```
Comment (rafpaf) [inline frontend/src/metabase/browse/components/ModelsTable.tsx:87]: We may not need this anymore either

File: frontend/src/metabase/browse/components/ModelsTable.tsx [hunk 2]
```diff
@@ -8,6 +8,7 @@ import { push } from "react-router-redux";
 import { t } from "ttag";
 
 import { getCollectionName } from "metabase/collections/utils";
+import { EllipsifiedPath } from "metabase/common/components/EllipsifiedPath/EllipsifiedPath";
```
Comment (iethree) [inline frontend/src/metabase/browse/components/ModelsTable.tsx:11]: we probably want an index.ts file so we can import as `metabase/common/components/EllipsifiedPath`

File: frontend/src/metabase/common/components/EllipsifiedPath/EllipsifiedPath.module.css [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  flex-shrink: 0;
+  padding-inline: 0.1rem;
+}
+
+.pathItem {
+  flex-grow: 1;
+  min-width: 0;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
```
Comment (iethree) [inline frontend/src/metabase/common/components/EllipsifiedPath/EllipsifiedPath.module.css:20]: i'm not seeing the ellipsis actually happening here.

Screen Shot 2024-08-06 at 10 19 50 AM

**Review by iethree — CHANGES_REQUESTED** [review]
I like the direction this is headed! some details to handle still

File: frontend/src/metabase/browse/components/ModelsTable.module.css [hunk 4]
```diff
@@ -0,0 +1,10 @@
+.collectionLink {
+  :hover {
+    &,
+    *,
+    *:before,
+    *:after {
+      color: var(--mb-color-brand) !important;
+    }
+  }
+}
```
Comment (iethree) [inline frontend/src/metabase/browse/components/ModelsTable.module.css:10]: this seems to work just fine? what does all that extra stuff do?

```suggestion
.collectionLink {
  :hover {
    color: var(--mb-color-brand);
  }
}
```

File: frontend/src/metabase/browse/components/ModelsTable.tsx (hunk 1 above)
Comment (iethree) [reply frontend/src/metabase/browse/components/ModelsTable.tsx:89]: I'm not seeing it even on a large dataset.  the skeletons are much nicer. go ahead and remove it.

File: frontend/src/metabase/common/components/EllipsifiedPath/EllipsifiedPath.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                {item}
+              </div>
+              {index < items.length - 1 && (
+                <div key={`${key}-sep`} className={S.slash}>
+                  /
+                </div>
+              )}
+            </>
+          );
+        })}
```
Comment (iethree) [inline frontend/src/metabase/common/components/EllipsifiedPath/EllipsifiedPath.tsx:48]: nit, if you generate the key inline you can eliminate some lines here
```suggestion
        {items.map((item, index) => (
            <>
              <div
                key={`${item}${index}`}
                ref={el => el && ref.current.set(key, el)}
                className={S.item}
              >
                {item}
              </div>
              {index < items.length - 1 && (
                <div key={`${key}-sep`} className={S.slash}>
                  /
                </div>
              )}
            </>
          ))}
```

File: frontend/src/metabase/common/components/EllipsifiedPath/EllipsifiedPath.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+type EllipsifiedPathProps = { items: string[]; tooltip: string };
+
+/**
+ * Displays a path such as "Collection / Subcollection / Subsubcollection /
+ * Parent Collection".
+ *
+ * If the path is too long to fit, some items may be truncated, like this:
+ * "Collection / Subcollec... / Subsub... / Parent Collection".
+ *
+ * The first and last item are never truncated.
```
Comment (iethree) [inline frontend/src/metabase/common/components/EllipsifiedPath/EllipsifiedPath.tsx:15]: not true!

File: frontend/src/metabase/common/components/EllipsifiedPath/EllipsifiedPath.tsx [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  const { areAnyTruncated, ref } = useAreAnyTruncated<HTMLDivElement>();
+
+  return (
+    <Tooltip label={tooltip} disabled={!areAnyTruncated} multiline maw="20rem">
+      <div className={S.path}>
+        {items.length > 1 && (
+          <div className={S.dots}>
+            … <div className={S.slash}>/</div>
+          </div>
+        )}
```
Comment (iethree) [inline frontend/src/metabase/common/components/EllipsifiedPath/EllipsifiedPath.tsx:29]: if we get to a narrow screen width where this block cuts off all leading paths, but the final collection name doesn't get truncated, we don't get a tooltip. this is a minor thing, I don't think it should block this PR, and I don't think we should increase the complexity of this component to handle it, but wanted to call it out.

Screen Shot 2024-08-14 at 9 44 01 AM

**Review by iethree — APPROVED** [review]
🥳  🐎 

Nice work. this is so much simpler and more performant than before.

**Pull request merged**